### PR TITLE
nrocli: validation fixes and retargeting

### PR DESCRIPTION
**Pull request by fromanirh**
enhance (and rename, since it's only supposed to validate) nrocli to detect NRT misconfiguration, even though only the most glaring.

Add along the way some fixes for the existing functionality, and options to be friendlier for integration and testing.
With these ehnancements, `nrovalidate` (formerly `nrocli`) can now detect and report incompatible environments, meaning environments on which NROP can be deployed but it will do nothing useful.

### CODE REVIEW

File: nrovalidate/main.go [hunk 1]
```diff
@@ -70,15 +91,11 @@ func parseArgs(args ...string) (ProgArgs, error) {
...
-	if pArgs.Version {
-		return pArgs, err
-	}
-
 	return pArgs, nil
 }
 
 func validateCluster(args ProgArgs) error {
-	cli, err := clientutil.New()
+	cli, err := NewClientWithScheme(scheme)
```
Comment (shajmakh) [inline nrovalidate/main.go:98]: what are the advantages of using the clientset with the scheme instead of clientset out of clientutil?

Comment (fromanirh) [reply nrovalidate/main.go:98]: we need to have extra schemas registered in the client (NRO, MCP) which aren't registered by default in the clientutil one.

**Comment by shajmakh** [comment]
/lgtm 
thanks!

File: .github/workflows/e2e-tools.yml [hunk 2]
```diff
@@ -45,7 +45,7 @@ jobs:
 
     - name: Build tools
       run: |
-        make build-tools binary-nrocli
+        make build-tools binary-nrovalidate
```
Comment (Tal-or) [inline .github/workflows/e2e-tools.yml:48]: it is a tool which supposes to validate, so maybe ```nrovalidator``` is more accurate here?

Comment (fromanirh) [reply .github/workflows/e2e-tools.yml:48]: I don't have strong feelings. We're not following too closely the unix conventions anyway, so if you want it renamed I can do it.

**Comment by Tal-or** [comment]
/lgtm
/approve
Thanks

**Pull request merged**